docs/src/main/asciidoc/kafka-streams-guide.adoc

```diff
@@ -0,0 +1,1141 @@
+                            .setScale(1, RoundingMode.HALF_UP)
+                            .doubleValue();
+
+                    LOG.info("station: {}, temperature: {}", station.name, temperature);
+                    return KafkaMessage.of(station.id, Instant.now() + ";" + temperature);
+                });
+    }
+
+    @Outgoing("weather-stations")                               // <3>
+    public Flowable<KafkaMessage<Integer, String>> weatherStations() {
+        List<KafkaMessage<Integer, String>> stationsAsJson = stations.stream()
+            .map(s -> KafkaMessage.of(s.id, "{ \"id\" : " + s.id + ", \"name\" : \"" + s.name + "\" }"))
+            .collect(Collectors.toList());
+
+        return Flowable.fromIterable(stationsAsJson);
+    };
+
+    private static class WeatherStation {
+
+        int id;
+        String name;
+        int averageTemperature;
+
+        public WeatherStation(int id, String name, int averageTemperature) {
+            this.id = id;
+            this.name = name;
+            this.averageTemperature = averageTemperature;
+        }
+    }
+}
+----
+<1> Instruct Reactive Messaging to dispatch the items from the returned `Flowable` to `temperature-values`.
+<2> The method returns a RX Java 2 _stream_ (`Flowable`) emitting a random temperature value every 0.5 seconds.
+<3> Instruct Reactive Messaging to dispatch the items from the returned `Flowable` (static list of weather stations) to `weather-stations`.
+
+The two method each return a _reactive stream_ whose items are sent to the streams named `temperature-values` and `weather-stations`, respectively.
+
+=== Topic Configuration
+
+The two streams are mapped to Kafka Apache topics using the Quarkus configuration file `application.properties`.
+For that, add the following to the file `producer/src/main/resources/application.properties`:
+
+[source]
+----
+kafka.bootstrap.servers=localhost:9092
+
+mp.messaging.outgoing.temperature-values.connector=smallrye-kafka
+mp.messaging.outgoing.temperature-values.key.serializer=org.apache.kafka.common.serialization.IntegerSerializer
+mp.messaging.outgoing.temperature-values.value.serializer=org.apache.kafka.common.serialization.StringSerializer
+
+mp.messaging.outgoing.weather-stations.connector=smallrye-kafka
+mp.messaging.outgoing.weather-stations.key.serializer=org.apache.kafka.common.serialization.IntegerSerializer
+mp.messaging.outgoing.weather-stations.value.serializer=org.apache.kafka.common.serialization.StringSerializer
+----
+
+This configures the Kafka bootstrap server, the two topics and the corresponding (de-)serializers.
+More details about the different configuration options are available on the https://kafka.apache.org/documentation/#producerconfigs[Producer configuration] and https://kafka.apache.org/documentation/#consumerconfigs[Consumer configuration] section from the Kafka documentation.
```

Comment:
Not really a comment, but do you know another place or a cheatsheet with the same content?

Reply:
No, I think that's the authoritative source.